ws: Support globs in configured origins

mindovermiles262 opened the pull request:
References #21738 #21727 

Changes the default matching behavior of cockpit `Origin` configuration to use fnmatch instead of string comparison, allowing for the use of wildcards and standard glob matching.

mindovermiles262 commented:
@martinpitt is there anything I need to do here?

martinpitt commented on src/websocket/websocketserver.c:
```diff
@@ -195,7 +196,7 @@ respond_handshake_rfc6455 (WebSocketServer *self,
         }
       for (i = 0; self->allowed_origins[i] != NULL; i++)
         {
-          if (g_ascii_strcasecmp (origin, self->allowed_origins[i]) == 0)
+          if (fnmatch (self->allowed_origins[i], origin, 0) == 0)
```
Please add `FNM_CASEFOLD` to keep the existing behaviour of ignoring case.

martinpitt reviewed (changes requested):
Hello @mindovermiles262 thanks for the ping! I'm not currently working on/subscribed to cockpit, so I didn't see the PR earlier. There's one thing to fix, and then please squash the commits, drop Signed-Off-By, and trim/fix the commit subject a bit and prefix it with "ws:" (like in "web server"):

> ws: Support globs in Origins 
>
> Use fnmatch() instead of precise string comparison.

This also needs an integration test in test/verify/check-connection , it can be added to testConfigOrigins(). If you are interested in doing that, https://github.com/cockpit-project/cockpit/blob/main/HACKING.md and https://github.com/cockpit-project/cockpit/blob/main/test/README.md describe how to build the project and run the tests. If not, I'm happy to write the integration test myself.

Thanks!

mindovermiles262 commented:
Hi @martinpitt  I think this fixes all of your feedback - I was able to figure out the testing myself, changed fnmatch to be case insensitive, and squashed my commits. Please let me know if there is anything else you would like me to do.

Thank you!

mindovermiles262 commented:
@martinpitt I'm not sure the error is anything on my end. Looks like maybe the builds are a bit flaky? Did you want to rerun it?

martinpitt commented:
@mindovermiles262 Nevermind, this is indeed a known flake. I'm just AFK over the Easter holidays. Thanks!
